searx/autocomplete.py

```diff
@@ -252,11 +253,36 @@ def yandex(query, _lang):
 }
 
 
-def search_autocomplete(backend_name, query, sxng_locale):
-    backend = backends.get(backend_name)
-    if backend is None:
-        return []
+def search_autocomplete(backend_names, query, sxng_locale):
+
+    enabled_backends = set()
+    for name in backend_names:
+        backend = backends.get(name)
+        if backend is None:
+            logger.error("%s is not valid! Must be one of %s", repr(backend), backend.keys())
+            continue
+        if backend not in enabled_backends:
+            enabled_backends.add(backend)
+
+    len_enabled_backends = len(enabled_backends)
+
     try:
-        return backend(query, sxng_locale)
+        results = []
+
+        for backend in enabled_backends:
+            backend_results = backend(query, sxng_locale)
+            if (len_enabled_backends > 1) and (len(backend_results) > 3):
+                # if more than 3 autocompleters: only get the first 3 results from each
+
+                results.extend(backend_results[:3:])
+
+            else:
+                results.extend(backend_results)
+
+        random.seed(1)
+        random.shuffle(results)
```

Comment:
We didn't shuffle results before that change either, I don't think we should start doing that now.

Reply:
Removed the shuffle but in the future the results must be somehow mixed.

Now the ordering in the settings matters as there can be more engines producing the following suggestions: 
- 3 from autocompleter1, then
- 3 from autocompleter2, then
- 3 from autocompleter3

Reply:
In the end, we should probably do
- result 1 from autocompleter 1
- result 1 from autocompleter 2
- result 1 from autocompleter 3
- ...
- result 2 from autocompler 1
- result 2 from autocompleter 2
- ...

with no more than 10 results/suggestions in total returned to the user.

At least that's the behavior I'd personally like the most as a user.